array-manipulation-1/src/compact.js

```diff
@@ -0,0 +1,12 @@
+/* exported compact */
+function compact(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length; i++) {
+
+    if (array[i] === null || array[i] === false || array[i] === undefined || array[i] === 0 || array[i] === '' || Object.is(array[i], NaN)) {
+      continue;
+    }
```

Comment:
You can test any value for falsiness like this
```js
if (!array[i]) {
  continue;
}
newArray.push(array[i]);
```

That way you don't need a big list of comparisons to check for any of these falsy values.